src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -372,6 +372,21 @@ module _ {ℓ} {R : Rel A ℓ} (R? : Decidable R) where
     x ∷ xs ++ y ∷ ys         ∎
     where open PermutationReasoning
 
+------------------------------------------------------------------------
+-- filter
+
+filter-↭ : ∀ {p} {P : Pred A p} (P? : Pred.Decidable P) → xs ↭ ys → filter P? xs ↭ filter P? ys
+filter-↭ P? refl = refl
+filter-↭ P? (prep x xs↭ys) with P? x
+... | yes _ = prep x (filter-↭ P? xs↭ys)
+... | no _  = filter-↭ P? xs↭ys
+filter-↭ P? (swap x y xs↭ys) with P? x in eqˣ | P? y in eqʸ
+... | yes _ | yes _ rewrite eqˣ rewrite eqʸ = swap x y (filter-↭ P? xs↭ys)
```

Comment:
I'm annoyed at having to match on `P? x` and `P? y` _twice_ each in each case to get the `filter` to normalize properly. Is there a way to avoid this?

Reply:
You could prove an unfolding lemma:
```
P x -> filter P? (x :: xs)  == x :: filter P? xs
```